src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient.cs

```diff
@@ -34,128 +35,177 @@ public ImageStoreClient(ServiceFabricHttpClient httpClient)
+                var files = Directory.EnumerateFiles(imageStorePath, "*", SearchOption.TopDirectoryOnly).Select(f => new System.IO.FileInfo(f)).Select(f => new Common.FileInfo(f.Length.ToString(), new FileVersion(), f.LastWriteTimeUtc, f.FullName.Replace(imageStorePath, ""))).ToList();
+                return new ImageStoreContent(files, directories);
             }
+            else
+            {
+                serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
+                var requestId = Guid.NewGuid().ToString();
+                var url = "ImageStore";
+                var queryParams = new List<string>();
 
-            return this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
+                // Append to queryParams if not null.
+                serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
+                queryParams.Add("api-version=6.0");
+                url += "?" + string.Join("&", queryParams);
+
+                HttpRequestMessage RequestFunc()
+                {
+                    var request = new HttpRequestMessage()
+                    {
+                        Method = HttpMethod.Get,
+                    };
+                    return request;
+                }
+
+                return await this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
+            }
         }
 
         /// <inheritdoc />
-        public Task CopyImageStoreContentAsync(
+        public async Task CopyImageStoreContentAsync(
             ImageStoreCopyDescription imageStoreCopyDescription,
             long? serverTimeout = 60,
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            imageStoreCopyDescription.ThrowIfNull(nameof(imageStoreCopyDescription));
-            serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
-            var requestId = Guid.NewGuid().ToString();
-            var url = "ImageStore/$/Copy";
-            var queryParams = new List<string>();
-            
-            // Append to queryParams if not null.
-            serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
-            queryParams.Add("api-version=6.0");
-            url += "?" + string.Join("&", queryParams);
-            
-            string content;
-            using (var sw = new StringWriter())
+            var store = await GetImageStoreConnectionString();
+            if (IsLocalStore(store))
             {
-                ImageStoreCopyDescriptionConverter.Serialize(new JsonTextWriter(sw), imageStoreCopyDescription);
-                content = sw.ToString();
-            }
 
-            HttpRequestMessage RequestFunc()
+                imageStoreCopyDescription.ThrowIfNull(nameof(imageStoreCopyDescription));
```

Comment:
>imageSt [](start = 16, length = 7)

Code to copy is missing. Please test with local file share.

Reply:
added, will have to do some manual tests since there is no tests in this repo